gateways/token_balances_api_gateway.py

```diff
@@ -0,0 +1,111 @@
+from typing import List, Optional
+
+from elasticsearch import Elasticsearch
+
+from common.configuration import ELASTIC_RESULTS_PER_PAGE, ELASTIC_SEARCH_TIMEOUT, ELASTIC_TOKEN_BALANCES_INDEX
+from gateways.clients.elastic_search_client import ElasticSearchClient
+from utils.elastic_search.elastic_search_utils import ElasticSearchUtils
+
+SORTABLE_FIELDS = {
+    'address': 'keyword',
+    'token_id': 'keyword',
+    'unlocked_balance': 'long',
+    'locked_balance': 'long',
+    'total': 'long',
+}
+
+
+class TokenBalancesApiGateway:
+    """ Gateway to interact with the Hathor ElasticSearch cluster with the token balances index
```

Comment:
I think adding a link to some documentation in Elasticsearch about how queries are built would be useful here for readers to easily learn more about what the code is doing.